contracts/BaseRouter.sol

```diff
@@ -232,7 +233,14 @@ abstract contract BaseRouter is Initializable {
                 _parseVaultIDs(actionsBorrow, dataBorrow, vaultManager);
                 _changeAllowance(IERC20(collateral), address(vaultManager), type(uint256).max);
                 _angleBorrower(vaultManager, actionsBorrow, dataBorrow, to, who, repayData);
-                _changeAllowance(IERC20(collateral), address(vaultManager), 0);
+            } else if (actions[i] == ActionType.addCollateralBorrower) {
+                (address collateral, address vaultManager, uint256 vaultID, uint256 amount) = abi.decode(
+                    data[i],
+                    (address, address, uint256, uint256)
+                );
+                if (amount == type(uint256).max) amount = IERC20(collateral).balanceOf(address(this));
```

Comment:
assuming all sweep actions went fine and there are not collat on the router, the caller has to do a transfer before calling this action